### PR TITLE
Update requirements.txt

### LINKED ISSUE #57: make run-server: "ModuleNotFoundError: No module named 'psycopg2'"
Environment:  Mac OS

```
python3 -m venv env
source env/bin/activate
pip install -r ./requirements.txt


$ make run-server
python ./kai/server.py
Traceback (most recent call last):
  File "/Users/jmatthews/git/jwmatthews/kai/./kai/server.py", line 26, in <module>
    from incident_store_advanced import Application, EmbeddingInstructor, PSQLIncidentStore
  File "/Users/jmatthews/git/jwmatthews/kai/kai/incident_store_advanced.py", line 14, in <module>
    import psycopg2
ModuleNotFoundError: No module named 'psycopg2'
make: *** [run-server] Error 1
```

### PR DESCRIPTION
**Pull request by jwmatthews**
Fixes #57 and #59

### CODE REVIEW

**Review by savitharaghunathan — APPROVED** [review]
ACK. Looks like the deps I installed locally

**Comment by JonahSussman** [comment]
Would strongly reccomend https://github.com/konveyor-ecosystem/kai/pull/62 instead. Will fix https://github.com/konveyor-ecosystem/kai/issues/60

File: requirements.txt [hunk 1]
```diff
@@ -106,23 +126,36 @@ pyzmq==25.1.1
...
+regex==2023.12.25
 requests==2.31.0
 rfc3339-validator==0.1.4
 rfc3986-validator==0.1.1
 rpds-py==0.10.6
+safetensors==0.4.2
+scikit-learn==1.4.1.post1
+scipy==1.12.0
 Send2Trash==1.8.2
+sentence-transformers==2.5.1
```
Comment (JonahSussman) [inline requirements.txt:138]: sentence-transformers must use `2.2.2`

**Review by JonahSussman — CHANGES_REQUESTED** [review]

**Comment by jwmatthews** [comment]
Close in favor of #62

**Pull request closed without merging**